src/x86_64/service.c

```diff
@@ -362,9 +424,24 @@ static void __attribute__((noinline)) init_service_new_stack()
 
     /* Switch to stage3 GDT64, enable TSS and free up initial map */
     init_debug("install GDT64 and TSS");
-    install_gdt64_and_tss();
+    install_gdt64_and_tss(0);
     unmap(PAGESIZE, INITIAL_MAP_SIZE - PAGESIZE, pages);
 
+#ifdef SMP_TEST
+    ipi_vector = allocate_interrupt();
+    assert(ipi_vector != INVALID_PHYSICAL);
+    register_interrupt(ipi_vector, closure(heap_general(kh), ipi_interrupt));
+
+    init_debug("performing SMP test");
+    idle_cpu_queue = allocate_queue(misc, 64);
+    start_cpu(misc, pages, TARGET_EXCLUSIVE_BROADCAST, new_cpu);
+
+    kernel_delay(seconds(1));
```

Comment:
I know this is just for testing, but I had a question about this in general. do we really ever need to barrier on 'all cpus available'? it seems like we can just let them trickle into the idle queue and pick up work? I guess that makes reporting the number of cpus difficult..but maybe that should come from the mptables?

Reply:
Absolutely. We have both the idle queue and the ci->online flag to tell us what cpus are available. This was just a test to see if they would all come up.